consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -746,7 +746,8 @@ impl PruningProofManager {
             .traversal_manager
             .anticone(pruning_point, self.virtual_stores.read().state.get().unwrap().parents.iter().copied(), None)
             .expect("no error is expected when max_traversal_allowed is None");
-        let anticone = self.ghostdag_managers[0].sort_blocks(anticone);
+        let mut anticone = self.ghostdag_managers[0].sort_blocks(anticone);
+        anticone.insert(0, pruning_point);
```

Comment:
in line 775 below I think you should `.take(self.ghostdag_k as usize + 1)`. Also, I think we should use the same reachability iterator method in `get_trusted_block_associated_ghostdagdata_block_hashes`. This also avoids the need to check for genesis, since we have origin as reachability and relations parent of genesis